### DISCUSSION TITLE
ProjectConfig Service: Enable access to $_appliedChanges

### DISCUSSION THREAD

**Discussion by MadMikeyB** [craftcms/cms#14851] [Ideas] [answered]
I'm working with the `EVENT_AFTER_WRITE_YAML_FILES` event (thanks to https://github.com/craftcms/cms/discussions/14365) for a plugin I am working on. 

I see that this event has access to an array called _appliedChanges, this array has loads of useful information but more importantly, a summary for changes for each project config change.

This is a `private` array and is defined within craft\services\ProjectConfig, and is mainly used within rememberAppliedChanges method: https://github.com/craftcms/cms/blob/4.8.8/src/services/ProjectConfig.php#L1693C21-L1726

For the purposes of making this event more helpful, it would be amazing to be able to access this event via a getter, e.g.

`$event->sender->getAppliedChanges()`

I'm hoping this would be as simple as a method such as the following within craft\services\ProjectConfig

```
    /**
     * Get the list of applied changes
     *
     * @return array
     */
    public function getAppliedChanges(): array
    {
        return $this->_appliedChanges;
    }
```

or making the $_appliedChanges array public? 

I have got some draft PR's in place with the above and if the P&T team do not object, I will mark them as ready for review.

Alternately, if you guys know of another way to access all of the handy messages detailing project config changes that you're outputting into this array it would be mega helpful.

**Comment by brandonkelly** [comment, marked as answer]
Thanks for the PRs! I’ve merged #14853 into `4.9`, and `4.9` into `5.1`.

**Reply by MadMikeyB** [reply]
Thanks Brandon, just checking this PR isn't needed fo 5.x? https://github.com/craftcms/cms/pull/14852

**Reply by brandonkelly** [reply]
Sorry, correct. Just closed the v5 PR.

**Reply by brandonkelly** [reply]
Craft 4.9 and 5.1 are out now with that change.

**Reply by MadMikeyB** [reply]
Thanks Brandon
